### ISSUE TITLE
Dependency classifier is not resolved for Ivy

### ISSUE THREAD

**Issue by teliatko** [coursier/coursier#461]
I have the dependency which is taken from Ivy repository but uses classifier, e.g.:

```
libraryDependencies ++= Seq(
  "co.oops" %% "useless" % "0.0.1",
  "co.oops" %% "useless" % "0.0.1" classifier "tests"
)
```

Dependency with `tests` classifier is not resolved, or to be correct it is but it is not correctly put in class path.

I've prepared repo for simulation of the issue at https://github.com/teliatko/coursier-issue-ivy-classifiers

Procedure after checkout is:
```
$ cd useless
$ sbt publishLocal
$ cd ../issue
$ sbt
> update
> show fullClasspath
```

Dependency `co.oops.useless` in on classpath only once and without classifier. Dependency tree produced by Coursier looks like this:

```
$ sbt
> coursierDependencyTree
issue
├─ co.oops:useless_2.11:0.0.1
|  ├─ org.scala-lang:scala-library:2.11.8
|  └─ org.scalactic:scalactic_2.11:3.0.1
|     ├─ org.scala-lang:scala-library:2.11.8
|     └─ org.scala-lang:scala-reflect:2.11.8
|        └─ org.scala-lang:scala-library:2.11.8
├─ co.oops:useless_2.11:0.0.1
|  ├─ org.scala-lang:scala-library:2.11.8
|  └─ org.scalactic:scalactic_2.11:3.0.1
└─ org.scala-lang:scala-library:2.11.8
```

It shows `co.oops:useless_2.11:0.0.1` twice and always without classifier `tests`. So `tests` classifier is somewhere lost along the way.

When I copy produced artifacts and pom into `~/.m2` and enable `Resolver.mavenLocal` in `issue/build.sbt` (of course leaving out `Resolver.defaultLocal`) everything works.

The `Resolver.defaultLocal` uses pattern (see SBT sources):

```
"[organisation]/[module]/" + PluginPattern + "[revision]/[type]s/[artifact](-[classifier]).[ext]"
```

Thus `classifier` part is in. Default SBT resolution works with `classifier` part correctly.

Side note: `issue` build already contains workaround from #444

**Comment by trentontrees** [comment]
I've got a similar looking issue.  The side note above #444 of `classpathTypes += "test-jar"` didn't work for my build.  Reverting back to 1.0.0-M15 did.  I'm curious if you have any links you would recommend for debugging sbt plugins.  Thanks in advance.

**Comment by caoilte** [comment]
Classifiers are icky! I just raised a similar issue for exclusion rules (#461)

**Comment by lustefaniak** [comment]
sbt-coursier doesn't resolve https://mvnrepository.com/artifact/com.sksamuel.elastic4s/elastic4s-testkit_2.11/2.3.1 dependency on 

```
<dependency>
<groupId>org.elasticsearch</groupId>
<artifactId>elasticsearch</artifactId>
<version>2.3.4</version>
<classifier>tests</classifier>
</dependency>
```

Which is required for embedded ES as it contains MockNode.class

**Comment by lustefaniak** [comment]
I'm not sure if I should add `classpathTypes += "test-jar"` to my project as it works just fine in default sbt configuration and I don't want to change anything which might have side effects just to make coursier happy.

**Comment by trentontrees** [comment]
I'm not able to use the `classpathTypes += "test-jar"` workaround in the `1.0.0-RC*` variants of cousier.  See #597 for working example

**Comment by nathankleyn** [comment]
We are also unable to move to coursier because of this bug. Hadoop publishes loads of test classified JARs, but most notably the `hadoop-common` and `hadoop-hdfs` JARs which contain (among other things) [the `MiniDFSCluster` class used for unit/integration testing Hadoop related things](https://wiki.apache.org/hadoop/HowToDevelopUnitTests).

```scala
"org.apache.hadoop" % "hadoop-hdfs" % "2.7.3" % "test" classifier "tests"
"org.apache.hadoop" % "hadoop-common" % "2.7.3" % "test" classifier "tests"
```

**Comment by reid-spencer** [comment]
+1

We also have this issue. In our case the classifier is "protobuf" for use with Google Protobuffers Sources

**Comment by zzvara** [comment]
Just moved to coursier for my project. I've got two classifiers as tests, but those are not put into class path. Resolved, however. What can we expect a quick fix?

**Comment by etorreborre** [comment]
It worked for me as well with coursier `1.0.0-RC6`.

**Comment by ikhoon** [comment]
I had the same problem. 
But this issue was fixed on [1.0.0-RC8](https://github.com/coursier/coursier/releases/tag/v1.0.0-RC8) without `classpathTypes += "test-jar"`.


:+1: :grinning:

**Comment by alexarchambault** [comment]
Yep, this should be fixed in `1.0.0-RC8` :-)
